src/PrpShop/PRP/Surface/QMipmap.cpp

```diff
@@ -298,332 +320,129 @@ QMipmap::QMipmap(plCreatable* pCre, QWidget* parent)
-                              QMessageBox::Ok);
-        return;
-    }
-    if (tex->isAlphaJPEG())
-        S.write(tex->getJpegAlphaSize(), tex->getJpegAlpha());
-    else
-        makeJAlphaSurface(tex, &S);
-    S.close();
+        try {
+            plDDSurface dds;
+            dds.setFromMipmap(tex);
+            dds.write(&S);
 
-    setExportDir(filename);
-}
+            S.close();
+        } catch (hsException& ex) {
+            QMessageBox::critical(this, tr("Error exporting DDS"),
+                                  QString::fromUtf8(ex.what()), QMessageBox::Ok);
+        }
+    } else if (file_ext == "jpg" || file_ext == "jpeg") {
+        hsFileStream S;
+        if (!S.open(qstr2st(filename), fmCreate)) {
+            QMessageBox::critical(this, tr("Error exporting JPEG"),
+                                  tr("Error: Could not open file %1 for writing").arg(filename),
+                                  QMessageBox::Ok);
+            return;
+        }
 
-void QMipmap::onImportDDS()
-{
-    plMipmap* tex = plMipmap::Convert(fCreatable);
-    QString filename = QFileDialog::getOpenFileName(this, tr("Import DDS"), getExportDir(),
-                                                    "DDS Files (*.dds)");
-    if (filename.isEmpty())
-        return;
+        if (tex->isImageJPEG()) {
+            S.write(tex->getJpegSize(), tex->getJpegImage());
+        } else {
+            size_t image_size = tex->GetUncompressedSize(0);
+            auto image_data = std::make_unique<unsigned char[]>(image_size);
+            tex->DecompressImage(0, image_data.get(), image_size);
+            plJPEG::CompressJPEG(&S, image_data.get(), image_size, tex->getWidth(), tex->getHeight(), tex->getBPP());
+        }
+    } else if (file_ext == "png") {
+        hsFileStream S;
+        if (!S.open(qstr2st(filename), fmCreate)) {
+            QMessageBox::critical(this, tr("Error exporting PNG"),
+                                  tr("Error: Could not open file %1 for writing").arg(filename),
+                                  QMessageBox::Ok);
+            return;
+        }
 
-    hsFileStream S;
-    if (!S.open(qstr2st(filename), fmRead)) {
-        QMessageBox::critical(this, tr("Error importing DDS"),
-                              tr("Error: Could not open file %1 for reading").arg(filename),
-                              QMessageBox::Ok);
-        return;
-    }
-    try {
-        plDDSurface dds;
-        dds.read(&S);
-        plMipmap* newTex = dds.createMipmap();
-        tex->CopyFrom(newTex);
-        delete newTex;
-    } catch (hsException& ex) {
-        QMessageBox::critical(this, tr("Error importing DDS"),
-                              QString::fromUtf8(ex.what()), QMessageBox::Ok);
+        try {
+            auto imageData = getTextureData(tex);
+            plPNG::CompressPNG(&S, imageData, tex->GetUncompressedSize(0), tex->getLevelWidth(0), tex->getLevelHeight(0), tex->getBPP());
```

Comment:
I think this will leak `imageData`. You might consider changing `getTextureData()` to return an `std::unique_ptr<char[]>`.